workers/price-reporter/src/reporter.rs

```diff
@@ -255,7 +196,7 @@ impl ConnectionMuxer {
 
                 // New price streamed from an exchange
```

Comment:
we should really break this loop up, control flow depth of 6 is pretty hard to read